### ISSUE TITLE
1.16.1 Repurposed Structure crashes with Structure Gel API mod

### ISSUE THREAD

**Issue by BentoGambin** [TelepathicGrunt/RepurposedStructures#17]
[crash-2020-08-03_21.49.28-client.txt](https://github.com/TelepathicGrunt/RepurposedStructures/files/5019509/crash-2020-08-03_21.49.28-client.txt)
Upon installing this mod, the game becomes unable to start, crashing right after forge's initial loading.
I don't know if it's an isolated issue (as my modlist is substanitial) or if it's common.
World was generated in 1.16 only.
Modlist:

- curios-FORGE-1.16.1-3.0
TrashSlot_1.16.1-12.0.1
curiouselytra-FORGE-1.16.1-3.0
Bountiful+Baubles+FORGE-1.16.1-0.0.1
industrial-foregoing-1.16.1-3.0.3-5935836
NetherPortalFix_1.16.1-7.0.0
titanium-1.16.1-3.0.2
movingelevators-1.2.17a-mc1.16
BetterMineshaftsForge-1.16.1-1.3
invtweaks-1.16.1-1.0.5
extradisks-1.16.1-1.3
Ding-1.16.1-1.1.0
curioofundying-FORGE-1.16.1-4.0
Controlling-7.0.0.2
AI-Improvements-1.16.1-0.3.0
caelus-FORGE-1.16.1-2.0-beta5
ImmersiveEngineering-0.16-109
guardvillagers-1.16.1-1.1.1
refinedstorage-1.9.2
structure-gel-api-1.16.1-1.1.1
dungeons-plus-1.16.1-1.0.2
BetterPingDisplay-1.16.1-1.0
PassableFoliage-1.16.1-2.0.0
JasonsDisenchantment_1.0.0_Forge_1.16.1
villagernames_1.16-2.4
BridgeMaker-1.16.1-1.0.0
fansmod-1.16.1-2.0.0.0-rc0
valhelsia_structures-16.0.1
BetterCaves-1.16.1-1.0
Variant16x-1.16-v0.13.3_beta
LanServerProperties-1.0
collective-1.16.1-1.35
ImprovedVanilla-1.16.1-1.1.0
MixinBootstrap-1.0.3
nopotionshift_1.16.1-1.5.0
FastLeafDecay-v23
CTM-MC1.16.1-1.1.0.1
DoubleSlabs-1.16-3.3.0
Regrowth-1.16.1-1.0.0.6
MekanismTools-1.16.1-10.0.8.431
Mekanism-1.16.1-10.0.8.431
MekanismGenerators-1.16.1-10.0.8.431
MekanismAdditions-1.16.1-10.0.8.431
JustEnoughResources-1.16.1-0.11.1.90
jei-1.16.1-7.0.1.9
Artifacts-1.16.1-2.5.2
DungeonCrawl-1.16.1-2.0.2
OptiFine_1.16.1_HD_U_G2
towers_of_the_wild-1.16.1-1.2.1
modnametooltip_1.16.1-1.14.0
corpse-1.16.1-1.0.8
itemcollectors-1.0.1-mc1.16

**Comment by TelepathicGrunt** [comment]
Well the crashlog says it is Optifine crashing which isn't surprising. I have zero clue why Optifine is suddenly unable to find Minecraft's Registry class with my mod but it didn't. It's pretty well known that due to Optifine being closed source and being a massive hack (due to replacing entire classes in Minecraft) that it often breaks other mods. 

But I never thought my own mod would break Optifine lmao. If you can show me the latest.log file in the log folder within the minecraft folder, it might offer some hints into why this crash occurs but I wouldn't hold my breath as this crashlog right now basically gives no info.

```
java.lang.NoClassDefFoundError: Could not initialize class net.minecraft.util.registry.Registry
	at net.optifine.util.BlockUtils.getBlock(BlockUtils.java:168) ~[?:?] {re:classloading}
	at net.optifine.config.ConnectedParser.parseBlockPart(ConnectedParser.java:254) ~[?:?] {re:classloading}
```

**Comment by TelepathicGrunt** [comment]
Also, try the latest forge version as well. maybe that might help

**Comment by nepphhh** [comment]
Getting the same crashes with brand new (.106) forge. These are very confusing crash logs. I initially thought it was these other mods (Morpheus or Flesh2Leather), as those were the ones indicated in the crash report. Looking at OP's log... maybe it turns out Optifine isn't to blame at all? The mod "blamed" seems fairly random to me. As soon as I removed one, it blamed another.

https://pastebin.com/BzNTBC2W
https://pastebin.com/06zpq1n7

**Comment by TelepathicGrunt** [comment]
What the. uhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhh

give me a day or so to poke other modders to help me figure this out lmao

**Comment by TelepathicGrunt** [comment]
Also, can you please get me the latest.log file instead of the crashlog file?

**Comment by TelepathicGrunt** [comment]
Actually I think I might know the issue. Give me a moment to make a test jar for you to test. if it works, I'll release it.

**Comment by TelepathicGrunt** [comment]
Here, can you try this jar and see if it stops the crash with other mods? Also, thank you for helping me by reporting this!

[repurposed_structures-1.16.1-2.0.1B.zip](https://github.com/TelepathicGrunt/RepurposedStructures/files/5019734/repurposed_structures-1.16.1-2.0.1B.zip)

**Comment by TelepathicGrunt** [comment]
Found the issue. It's that Repurposed Structures will crash if Structure Gel API is on. I'll try and figure out where the conflict lies with that mod. 

Again, thank you for bringing this to my attention! For now, you can either remove Structure Gel API or remove my mod. I'll leave up this issue report for others to see until I make a fix on my end to stop the conflict

**Comment by nepphhh** [comment]
Thanks! I appreciate you taking the time to track this one down.

**Comment by TelepathicGrunt** [comment]
No problem!
I left an issue report for them as well so hopefully a fix for either one of us will be made soon.
https://gitlab.com/modding-legacy/structure-gel-api/-/issues/1

**Comment by TelepathicGrunt** [comment]
Fixed in v2.0.1 now! I moved my mixin slightly earlier to avoid their javascript coremod.
https://www.curseforge.com/minecraft/mc-mods/repurposed-structures/files/3023687

Thank you for reporting this bug!
